core/src/main/java/com/scalar/db/storage/cassandra/CassandraAdmin.java

```diff
@@ -1,69 +1,278 @@
 package com.scalar.db.storage.cassandra;
 
+import static com.scalar.db.util.Utility.getFullNamespaceName;
+import static com.scalar.db.util.Utility.getFullTableName;
+
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import com.datastax.driver.core.schemabuilder.Create;
+import com.datastax.driver.core.schemabuilder.CreateKeyspace;
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
+import com.datastax.driver.core.schemabuilder.SchemaBuilder.Direction;
+import com.datastax.driver.core.schemabuilder.SchemaStatement;
+import com.datastax.driver.core.schemabuilder.TableOptions.CompactionOptions;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.inject.Inject;
 import com.scalar.db.api.DistributedStorageAdmin;
+import com.scalar.db.api.Scan.Ordering.Order;
 import com.scalar.db.api.TableMetadata;
 import com.scalar.db.config.DatabaseConfig;
 import com.scalar.db.exception.storage.ExecutionException;
 import com.scalar.db.exception.storage.StorageRuntimeException;
+import com.scalar.db.io.DataType;
+import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import javax.annotation.concurrent.ThreadSafe;
 
 @ThreadSafe
 public class CassandraAdmin implements DistributedStorageAdmin {
 
+  public static final String NETWORK_STRATEGY = "network-strategy";
+  public static final String COMPACTION_STRATEGY = "compaction-strategy";
+  public static final String REPLICATION_FACTOR = "replication-factor";
   private final ClusterManager clusterManager;
   private final CassandraTableMetadataManager metadataManager;
-  private final Optional<String> namespacePrefix;
+  private final Optional<String> keyspacePrefix;
 
   @Inject
   public CassandraAdmin(DatabaseConfig config) {
     clusterManager = new ClusterManager(config);
-    metadataManager = new CassandraTableMetadataManager(clusterManager);
-    namespacePrefix = config.getNamespacePrefix();
+    metadataManager =
+        new CassandraTableMetadataManager(clusterManager, config.getNamespacePrefix());
+    keyspacePrefix = config.getNamespacePrefix();
   }
 
   @VisibleForTesting
-  CassandraAdmin(CassandraTableMetadataManager metadataManager, Optional<String> namespacePrefix) {
-    clusterManager = null;
+  CassandraAdmin(
+      CassandraTableMetadataManager metadataManager,
+      ClusterManager clusterManager,
+      Optional<String> namespacePrefix) {
+    this.clusterManager = clusterManager;
     this.metadataManager = metadataManager;
-    this.namespacePrefix = namespacePrefix.map(n -> n + "_");
+    this.keyspacePrefix = namespacePrefix;
   }
 
   @Override
   public void createTable(
       String namespace, String table, TableMetadata metadata, Map<String, String> options)
       throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+    String fullKeyspace = fullKeyspace(namespace);
+    createKeyspace(fullKeyspace, options);
```

Comment:
Sorry. One more minor suggestion:
```suggestion
    createKeyspaceIfNotExists(fullKeyspace, options);
```

Reply:
Indeed Thank you for all the detailed suggestions.